Makefile

```diff
@@ -198,12 +198,12 @@ test: vendor $(ADDON_ASSET_FILE)
 .PHONY: integration
 integration: $(MINISHIFT_BINARY)
 	mkdir -p $(INTEGRATION_TEST_DIR)
-	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) --run-before-feature="$(RUN_BEFORE_FEATURE)" --tags basic $(GODOG_OPTS)
+	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) --run-before-feature="$(RUN_BEFORE_FEATURE)" --tags basic "$(GODOG_OPTS)"
```

Comment:
@agajdosi Any specific reason for this change? It was working fine previously, right?

Reply:
@LalatenduMohanty It was working, because we did not use multiple tags concatenated with multiple "&&" operands. In this case all the tags between "&&"s got ignored. Due to this I have put it into brackets, which prevents this issue.

Reply:
@LalatenduMohanty Another possibility would be to use just "positive" tags and keep the negative just for `cdk-only` or `minishift-only`. Maybe this would be more readable. WDYT?

Reply:
-1, afaict this breaks the use of GODOG_OPTS completely. I actually created an issue - https://github.com/minishift/minishift/issues/1679, but it turns out that the quotes cause the problem. As soon as you provide GODOG_OPTS things fall apart. For example:

    $ make integration GODOG_OPTS="--tags addon-xpaas"

    flag provided but not defined: -tags addon-xpaas
    Usage of /var/folders/39/70n784613gd370g63ggg62lh0000gn/T/    go-build708633840/github.com/minishift/minishift/test/    integration/_test/integration.test:
      -binary string
        	Path to minishift binary
      -definitions
    
      -format string
        	Sets which format godog will use (default "pretty")
      -minishift-args string
        	Arguments to pass to minishift
      -no-colors
        	Disable colors in godog output
      -paths string
        	 (default "./features")
      -run-before-feature string
        	Set of minishift commands to be executed before every feature.     Individual commands must be delimited by a semicolon.
      -stop-on-failure
        	Stop when failure is found
      -tags string
        	Tags for godog test
      -test-dir string
        	Path to the directory in which to execute the tests
      -test.bench regexp
        	run only benchmarks matching regexp
      -test.benchmem
        	print memory allocations for benchmarks
      -test.benchtime d
        	run each benchmark for duration d (default 1s)
      -test.blockprofile file
        	write a goroutine blocking profile to file
      -test.blockprofilerate rate
        	set blocking profile rate (see runtime.SetBlockProfileRate) (    default 1)
      -test.count n
        	run tests and benchmarks n times (default 1)
      -test.coverprofile file
        	write a coverage profile to file
      -test.cpu list
        	comma-separated list of cpu counts to run each test with
      -test.cpuprofile file
        	write a cpu profile to file
      -test.list regexp
        	list tests, examples, and benchmarch maching regexp then exit
      -test.memprofile file
        	write a memory profile to file
      -test.memprofilerate rate
        	set memory profiling rate (see runtime.MemProfileRate)
      -test.mutexprofile string
        	write a mutex contention profile to the named file after     execution
      -test.mutexprofilefraction int
        	if >= 0, calls runtime.SetMutexProfileFraction() (default 1)
      -test.outputdir dir
        	write profiles to dir
      -test.parallel n
        	run at most n tests in parallel (default 4)
      -test.run regexp
        	run only tests and examples matching regexp
      -test.short
        	run smaller test suite to save time
      -test.timeout d
        	panic test binary after duration d (0 means unlimited)
      -test.trace file
        	write an execution trace to file
      -test.v
        	verbose: print additional output
    exit status 2
    FAIL	github.com/minishift/minishift/test/integration	0.067s
    make: *** [integration] Error 1    

I think some work is needed here. Documentation-wise as well as how the integration targets work. Maybe we should actually re-open #1679.